src/utilities/qobjectlistmodel.cpp

```diff
@@ -144,6 +145,50 @@ void QObjectListModel::setList(QList<QObject *> *list)
     delete oldList;
 }
 
+void QObjectListModel::synchronizeList(const QList<QObject *> &list)
+{
+    ::synchronizeList(this, *_list, list);
+}
+
+int QObjectListModel::insertRange(int index, int count, const QList<QObject *> &source, int sourceIndex)
+{
+    const int end = index + count - 1;
+    beginInsertRows(QModelIndex(), index, end);
+
+    QList<QObject *> inserted;
+    for (int i = 0; i < count; ++i) {
+        QObject *item(source.at(sourceIndex + i));
+        _list->insert(index + i, item);
+        inserted.append(item);
+    }
+    endInsertRows();
+
+    foreach (QObject *item, inserted) {
+        emit itemAdded(item);
+    }
+    emit itemCountChanged();
+    return end - index + 1;
+}
+
+int QObjectListModel::removeRange(int index, int count)
+{
+    const int end = index + count - 1;
+    beginRemoveRows(QModelIndex(), index, end);
+
+    QList<QObject *> removed;
+    for (int i = 0; i < count; ++i) {
+        removed.append(_list->at(index));
+        _list->removeAt(index);
+    }
+    endRemoveRows();
+
+    foreach (QObject *item, removed) {
+        emit itemRemoved(item);
```

Comment:
Is it possible someone connected to this signal would delete the object?  If so it would be safer to persist the removed and inserted lists over the duration of synchronizeLists and remove items from the opposing list and only emit the itemRemoved and itemInserted signals after synchronization has completed.